### ISSUE TITLE
namespaced packages break `yarn cache ls`

### ISSUE THREAD

**Issue by AsaAyers** [yarnpkg/yarn#1569]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

```
yarn cache v0.16.1
error An unexpected error occured, please open a bug report with the information provided in "/tmp/foo/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/cache for documentation about this command.
```

The end of that log file is:

```
Trace: 
  Error: ENOENT: no such file or directory, open '/home/asa/.yarn-cache/npm-@kadira/.yarn-metadata.json'
      at Error (native)
```
I looked at `~/.yarn-cache/npm-@kadira` and it contains several packages:

```
react-split-pane-1.4.7         storybook-addon-links-1.0.1  storybook-channel-postmsg-1.1.0
storybook-2.28.1               storybook-addons-1.6.1       storybook-ui-3.6.3
storybook-addon-actions-1.1.1  storybook-channel-1.1.0
```

**If the current behavior is a bug, please provide the steps to reproduce.**

1. `yarn cache clean`
2. `yarn cache ls`
  * to verify it's working
5. `yarn add @kadira/storybook`
6. `yarn cache ls`
  * This will error

**What is the expected behavior?**

It should list what's in the cache.

**Please mention your node.js, yarn and operating system version.**

```
asa@asa:/tmp/foo$ node --version
v6.2.0
asa@asa:/tmp/foo$ yarn --version
0.16.1
asa@asa:/tmp/foo$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"
```

**Comment by kittens** [comment]
Fixed in #1279.

**Comment by AsaAyers** [comment]
Thanks :) I searched for "namespaced" and didn't think to search for "scoped"
